Src/Support/GoogleApis/Apis/[Media]/Upload/ResumableUpload.cs

```diff
@@ -390,22 +426,81 @@ public Task<IUploadProgress> UploadAsync()
-        /// <param name="cancellationToken">A cancellation token to cancel operation.</param>
-        public async Task<IUploadProgress> ResumeAsync(CancellationToken cancellationToken)
+        /// <summary>
+        /// <para>Asynchronously resumes the upload from the last point it was interrupted.</para>
+        /// <para>Use when resuming and the program was not restarted.</para>
+        /// <para>You do not need to seek to restart point in the media file.</para>
+        /// </summary>
+        /// <param name="cancellationToken">A cancellation token to cancel the asynchronous operation.</param>
+        public Task<IUploadProgress> ResumeAsync(CancellationToken cancellationToken)
         {
+            return ResumeAsync(cancellationToken, null);
+        }
+        /// <summary>
+        /// <para>Asynchronously resumes the upload from the last point it was interrupted.</para>
+        /// <para>Use when resuming and the program was restarted.</para>
+        /// <para>In your application's UploadSessionData Event Handler, store UploadUri.AbsoluteUri property value (resumable session URI string value)
+        /// <para>to persistent storage for use with Resume() or ResumeAsync() upon a program restart.</para>
+        /// <para>It is strongly recommended that the FullPathFilename of the media file that is being uploaded is saved also so that a subsequent execution of the</para>
+        /// <para>program can compare the saved FullPathFilename value to the FullPathFilename of the media file that it has opened for uploading.</para>
+        /// <para>You do not need to seek to restart point in the media file.</para>
+        /// </summary>
+        /// <param name="uploadUri">VideosResource.InsertMediaUpload UploadUri property value that was saved to persistent storage during a prior execution.</param>
+        public Task<IUploadProgress> ResumeAsync(Uri uploadUri)
+        {
+            return ResumeAsync(CancellationToken.None, uploadUri);
+        }
+        /// <summary>
+        /// <para>Asynchronously resumes the upload from the last point it was interrupted.</para>
+        /// <para>Use when the program was restarted and you wish to resume the upload that was in progress when the program was halted.</para>
+        /// <para>In your application's UploadSessionData Event Handler, store UploadUri.AbsoluteUri property value (resumable session URI string value)
+        /// <para>to persistent storage for use with Resume() or ResumeAsync() upon a program restart.</para>
+        /// <para>It is strongly recommended that the FullPathFilename of the media file that is being uploaded is saved also so that a subsequent execution of the</para>
+        /// <para>program can compare the saved FullPathFilename value to the FullPathFilename of the media file that it has opened for uploading.</para>
+        /// <para>You do not need to seek to restart point in the media file.</para>

```

Comment:
By "media file" do you mean the contentStream passed to the constructor?
Where is the seek to the correct location  in ContentStream done? I might be missing something but I can't see it.